### PR TITLE
Updating occurrence examples to avoid confusion

### LINKED ISSUE #311: Filters reference should avoid the use of `occurrence` examples
The filters reference doc contains a number of examples which refer to the event `occurrences` attribute. These examples are illustrative of some filter capabilities, but now that we ship a built-in `occurrences`  filter extension, there's more opportunity for confusion about how these things are related.

IMO we should replace the examples in the filter doc with equally illustrative ones that do not rely on the `occurrences` attribute.

### PR DESCRIPTION
**Pull request by asachs01**
## Description
Fixes #311 

## Motivation and Context
Removes confusing `occurrence` examples.

## How Has This Been Tested?
`yarn`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New doc/guide
- [x] Fixing errata

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] All tests have passed.

### CODE REVIEW

**Comment by cwjohnston** [comment]
LGTM so far. I think this filter needs testing prior to shipping these changes. 🤔

File: content/sensu-core/0.29/reference/filters.md [hunk 1]
```diff
@@ -245,26 +245,25 @@ attributes][8] (including custom attributes).
...
-of `60`.
+token (`:::client.environent|known:::`) as a secondary attribute in the Ruby eval
+expression. The token will be replaced by the [custom client definition attribute][5]
+named `environment` if it is defined, otherwise it will use the fallback value
+of `unknown`.
 
 {{< highlight json >}}
 {
   "filters": {
-    "occurrences": {
```
Comment (cwjohnston) [inline content/sensu-core/0.29/reference/filters.md:256]: I had to change the filter to make it work:

```
{
  "filters": {
    "pre_prod": {
      "negate": true,
      "attributes": {
        "client": {
          "environment": "eval: ['qa','staging','test'].include? ':::client.environment|unknown:::'"
        }
      }
    }
  }
}
```

Note the single quotes around the token substitution.

**Review by cwjohnston — CHANGES_REQUESTED** [review]

**Comment by majormoses** [comment]
I don't think we should remove the examples of occurrences and instead should include snippets telling users that they must include the occurences filter on their handler for it to take effect.

File: content/sensu-core/0.29/reference/filters.md [hunk 2]
```diff
@@ -245,26 +245,25 @@ attributes][8] (including custom attributes).
 ### Example filter attribute eval token
 
 The following is an example Sensu [filter definition][2], which is using a
-token (`:::check.occurrences|60:::`) as a secondary attribute in the Ruby eval
-expression. The token will be replaced by the [check definition attribute][6]
-named `occurrences` if it is defined, otherwise it will use the fallback value
-of `60`.
+token (`:::client.environent|known:::`) as a secondary attribute in the Ruby eval
```
Comment (cwjohnston) [inline content/sensu-core/0.29/reference/filters.md]: I think we want `:::client.environment|known:::` to be `:::client.environment|unknown:::` for tying into our example filter

**Review by cwjohnston — APPROVED** [review]

**Pull request merged**